### DISCUSSION TITLE
Enable using images from a url or local directory

### DISCUSSION THREAD

**Discussion by kbaran1998** [AntoniosBarotsis/qr-rs#9] [Ideas]
## Problem
Currently, it is only possible for the google logo to be the logo in the overlay.

## Possible Solution
It would be nice to be able to point to a directory or a web link to an image for option `-l, --logo <LOGO>` in the CLI and sever. The command could look something like this:

### Local dir
```shell
qrg github.com -d tmp.png -l ./github.png
```

### Web link
```shell
qrg github.com -d tmp.png -l https://cdn-icons-png.flaticon.com/512/25/25231.png
```

**Comment by AntoniosBarotsis** [comment]
My initial thought process was that there are only so many logos that we would possibly want to use for this, say Google's logo and then the social media icons from platforms we use so I opted for the approach I have now which is simpler (and safer) in the code. The images are included in the resulting binary, making the whole project very portable.

At the same time, it would be nice to support this as, in the case that we needed a new logo, I'd have to make a new release right now to include it right now.

I'll have to think a bit about how to actually implement this as in, should the same parameter be used for both links, paths and the integrated logos? I feel like it would be better if I made a separate parameter `--logo-source` or something like that which can be a link (server+cli) or a path to a file (cli only). 

What do you think?

**Reply by kbaran1998** [reply]
Yeah, I agree, we can have a limited number of logos as `assets` which should be SVGs (as they scale with the image well, given the fact that this tool has a `-s, --size <SIZE>` option). Also, I think the `--logo-source` param would be a good solution.

**Reply by AntoniosBarotsis** [reply]
Good point on the SVGs as well, I'll be implementing it like this then, thanks for the suggestion!

**Reply by AntoniosBarotsis** [reply]
Small note on this, I just started looking into it.

The image processing library that I am using is focused on raster graphics and does not support SVGs. There likely is a workaround where I parse it with a different library and convert it into a format that the former understands but in second thoughts, SVGs don't make *too* much sense since the logo in the middle is small anyway. It is a bit inconvenient because if you have SVG images saved you'd need to convert them but practically it should look fine. 

What do you think @kbaran1998?
